src/input.rs

```diff
@@ -0,0 +1,216 @@
+//! Input processing functionality for Sprocket.
+
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::bail;
+use tracing::debug;
+
+/// Parse result containing either JSON or YAML data
+#[derive(Debug)]
+pub enum ParsedInput {
+    /// JSON data
+    Json(serde_json::Value),
+    /// YAML data converted to JSON
+    Yaml(serde_json::Value),
+}
```

Comment:
I'm not convinced this enum should hold a JSON `Value`. It's important to think about ownership of data, and this `enum` seems to be doing two unrelated tasks at once.
1. It is tracking what format the input file was in
2. It is owning the JSON value we want to pass into `validate_inputs()`

IMO those two tasks should be done separately.

Reply:
@Gyan-max this has not been resolved. It should look something like:
```suggestion
pub enum InputFormat {
    /// JSON
    Json,
    /// YAML
    Yaml,
}
```